src/commands/retrieve_token.rs

```diff
@@ -0,0 +1,172 @@
+use crate::authenticator::HttpAuthenticator;
+use crate::http_client::HttpClient;
+use crate::jwt::signer::JwtSignerImpl;
+use crate::jwt::signer::local::LocalPrivateKeySigner;
+use crate::system_identity::input_data::SystemTokenCreationMetadata;
+use crate::system_identity::input_data::auth_method::AuthMethod;
+use crate::token::Token;
+use crate::token_retriever::TokenRetrieverWithCache;
+use crate::{TokenRetriever, TokenRetrieverError};
+use std::sync::Arc;
+
+pub struct RetrieveTokenCommand<C>
+where
+    C: HttpClient,
+{
+    http_client: Arc<C>,
+}
+
+impl<C> RetrieveTokenCommand<C>
+where
+    C: HttpClient,
+{
+    pub fn new(http_client: Arc<C>) -> Self {
+        Self { http_client }
+    }
+
+    pub fn retrieve_token(
+        &self,
+        metadata: &SystemTokenCreationMetadata,
+    ) -> Result<Token, TokenRetrieverError> {
+        let http_authenticator = HttpAuthenticator::new(
+            self.http_client.clone(),
+            metadata.environment.token_renewal_endpoint(),
+        );
```

Comment:
Would it make sense to keep the `http_authenticator` as a field in `RetrieveTokenCommand` instead of the `HttpClient`? We wouldn't need to clone the `http_client` and we may be able to avoid the whole `Arc` thing.